### PR TITLE
User can Sign up for an account to get access to buying a subscription

### PR DESCRIPTION
**Pull request by dernathul**
# [PT Story](https://www.pivotaltracker.com/story/show/172048734)

```
As a user
In order to create my account and get access to buying a subscription
I would like to be able so sign up an account
```
## Changes proposed in this pull request:
* Add form for signing up, so you can access to the subscription form.
* Test for happy and sad paths
* Semantic UI used for some styling in the form of sign up and login.
## What I have learned working on this feature:
* To implement signup form using J-tockauth
* Refreshed knowledge about TDD

## Screenshots
Screenshot 2020-04-02 at 10 51 02
Screenshot 2020-04-02 at 10 51 14

### CODE REVIEW

File: cypress/fixtures/sign_up.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "id": 3,
+    "email": "john-doe1@gmail.com",
+    "provider": "email",
+    "uid": "john-doe1@gmail.com",
+    "role": "reg_user",
+    "allow_password_change": false
+  }
+}
+
+
```
Comment (emtalen) [inline cypress/fixtures/sign_up.json]: You should probably have the response as the response comes, so it should be data.data as keys. and remove extra lines :)

File: src/components/LoginForm.jsx [hunk 2]
```diff
@@ -20,20 +20,21 @@ const LoginForm = props => {
     } catch (error) {
       console.log(error);
     }
-  };
+  };debugger
```
Comment (pgaunitz) [inline src/components/LoginForm.jsx]: remove debugger

File: src/components/LoginForm.jsx [hunk 3]
```diff
@@ -20,20 +20,21 @@ const LoginForm = props => {
...
```
Comment (emtalen) [inline src/components/LoginForm.jsx]: remove debugger

File: src/components/SignUpForm.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      });
+      props.dispatch({
+        type: AUTHENTICATE,
+        payload: {
+          currentUser: { email: response.data.data.email, role: response.data.data.role }
+        }
+      });
+    } catch (error) {
+      console.log(error);
+    }debugger
```
Comment (pgaunitz) [inline src/components/SignUpForm.jsx]: remove debugger

File: src/components/SignUpForm.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          type="password"
+          placeholder="Confirm Password"
+        />
+        <button id="signup-button" type="signup">
+          Sign Up
+        </button>
+      </form>
+    );
+  }
+  debugger;
```
Comment (pgaunitz) [inline src/components/SignUpForm.jsx]: remove debugger

File: src/components/SignUpForm.jsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
```
Comment (emtalen) [inline src/components/SignUpForm.jsx:71]: remove debugger

File: src/index.css [hunk 7]
```diff
@@ -1,6 +1,10 @@
...
-    margin-left: 21cm;
-    font-size: large;
+#logout-button {
+    margin-left:  20.5cm
+}
+
+#sign-up {
+    margin-left: 20cm;
+    
 }
```
Comment (emtalen) [inline src/index.css:7]: is this responsive?

File: src/components/CategoryHeader.jsx [hunk 8]
```diff
@@ -38,7 +39,19 @@ const CategoryHeader = props => {
...
+        props.dispatch({
+          type: SHOW_SIGN_UP_FORM,
+          payload: { showSignUpForm: true }
+        })
+      }
+    >
+      Sign up
+    </button>
+    <button
           class="ui secondary button"
```
Comment (emtalen) [inline src/components/CategoryHeader.jsx:55]: formatting

File: src/components/LoginForm.jsx [hunk 9]
```diff
@@ -20,20 +20,21 @@ const LoginForm = props => {
...
 
   const onLogout = () => {
     auth.signOut();
   };
-
+  
   let login;
   if (props.authenticated) {
+    let  cutEmail = props.currentUser.email.substring(0, props.currentUser.email.indexOf('@'))
```
Comment (emtalen) [inline src/components/LoginForm.jsx]: Nicely done with this one! 💯

**Review by pgaunitz — CHANGES_REQUESTED** [review]
nice with the split email

File: src/components/LoginForm.jsx (hunk 9 above)
Comment (dernathul) [reply src/components/LoginForm.jsx]: mr janko doing

Comment (janko101) [reply src/components/LoginForm.jsx]: Thank you :)

File: src/components/LoginForm.jsx (hunk 2 above)
Comment (janko101) [reply src/components/LoginForm.jsx]: removed

File: src/components/SignUpForm.jsx (hunk 4 above)
Comment (janko101) [reply src/components/SignUpForm.jsx]: removed

File: cypress/fixtures/sign_up.json (hunk 1 above)
Comment (janko101) [reply cypress/fixtures/sign_up.json]: did it

File: src/index.css (hunk 7 above)
Comment (Carltesio) [reply src/index.css:7]: We can fix this in the styling. Next feature

File: src/components/CategoryHeader.jsx (hunk 8 above)
Comment (janko101) [reply src/components/CategoryHeader.jsx:55]: fixed

**Review by emtalen — APPROVED** [review]

**Review by pgaunitz — APPROVED** [review]

**Pull request merged**